physx/src/scene.rs

```diff
@@ -517,6 +519,15 @@ impl SceneBuilder {
         self
     }
 
+    /// Sets a userdata pointer that will be carried into the filter shader callback function,
+    /// through the constantBlock parameter.
+    ///
+    /// Default: null
+    pub fn set_filter_shader_userdata(&mut self, user_data: *mut std::ffi::c_void) -> &mut Self {
+        self.filter_shader_user_data = user_data;
```

Comment:
This feels like a footgun. This *should* probably be both Sync and Pinned for safety reasons, but I'm not sure how we can do that.

Reply:
Total footgun. When I needed something similar before I put something in a Box and returned a pointer to that, it worked fine though :P